cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/SecureSystemReader.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.Resources;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Set;
+import org.eclipse.jgit.errors.ConfigInvalidException;
+import org.eclipse.jgit.lib.Config;
+import org.eclipse.jgit.storage.file.FileBasedConfig;
+import org.eclipse.jgit.util.FS;
+import org.eclipse.jgit.util.SystemReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link SystemReader} that disables features and hides properties to ensure
+ * user code isn't executed accidentally.
+ */
+public class SecureSystemReader extends SystemReader {
+
+  private static final String HOSTNAME = "cdap.io";
+  private static final String GIT_CONFIG_FILE_NAME = "/git/config";
+  private static final Logger LOG = LoggerFactory.getLogger(
+      SecureSystemReader.class);
+  // ALLOWED_SYSTEM_PROPERTIES stores properties that are essential for git to
+  // work.
+  private static final Set<String> REQUIRED_SYSTEM_PROPERTIES =
+      Collections.singleton("os.name");
+
+  private final Config gitConfig;
+  private final SystemReader delegatee;
+
+  private SecureSystemReader(SystemReader delagatee) {
+    this.delegatee = delagatee;
+    this.gitConfig = new Config();
+    try {
+      ClassLoader cl = this.getClass().getClassLoader();
+      URL url = this.getClass().getResource(GIT_CONFIG_FILE_NAME);
```

Comment:
Changed. The idea was that having the gitconfig stored as a file would allow people to go through it faster. Since we have only one property being set, I think both ways are good.